### PR TITLE
Raise from `try decorator`

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Raise from `try decorator` when using `send_signed_transaction`.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Replaces #87

### CODE REVIEW

File: plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py [hunk 1]
```diff
@@ -1025,15 +1025,19 @@ def send_signed_transaction(self, tx_signed: JSONLike) -> Optional[str]:
         :param tx_signed: the signed transaction
         :return: tx_digest, if present
         """
-        tx_digest = self._try_send_signed_transaction(tx_signed)
+        tx_digest = self._try_send_signed_transaction(tx_signed, raise_on_try=True)
```
Comment (DavidMinarsch) [inline plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py:1028]: raise_on_try should be a kwarg of send_signed_transaction and default to False. Then in the ledger connection we can pass the appropriate keyword

Comment (Adamantios) [reply plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py]: Done in 8d0f35b107d57b7674129f31ce47b1f85e67ff9d

File: packages/fetchai/connections/ledger/ledger_dispatcher.py [hunk 2]
```diff
@@ -277,25 +277,28 @@ def send_signed_transaction(
         :param dialogue: the dialogue
         :return: the ledger api message
         """
-        transaction_digest = api.send_signed_transaction(
-            message.signed_transaction.body
-        )
+        try:
+            transaction_digest = api.send_signed_transaction(
+                message.signed_transaction.body
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/ledger/ledger_dispatcher.py:282]: raise_on_try = True missing?

**Review by DavidMinarsch — COMMENTED** [review]
Small issue - o/w lgtm

File: packages/fetchai/connections/ledger/ledger_dispatcher.py (hunk 2 above)
Comment (Adamantios) [reply packages/fetchai/connections/ledger/ledger_dispatcher.py:282]: :man_facepalming:

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**